src/execution/operator/projection/physical_tableinout_function.cpp

```diff
@@ -52,15 +56,152 @@ unique_ptr<GlobalOperatorState> PhysicalTableInOutFunction::GetGlobalOperatorSta
 	return std::move(result);
 }
 
-OperatorResultType PhysicalTableInOutFunction::Execute(ExecutionContext &context, DataChunk &input, DataChunk &chunk,
-                                                       GlobalOperatorState &gstate_p, OperatorState &state_p) const {
-	auto &gstate = (TableInOutGlobalState &)gstate_p;
-	auto &state = (TableInOutLocalState &)state_p;
-	TableFunctionInput data(bind_data.get(), state.local_state.get(), gstate.global_state.get());
-	if (projected_input.empty()) {
-		// straightforward case - no need to project input
-		return function.in_out_function(context, data, input, chunk);
+void PhysicalTableInOutFunction::AddProjectedColumnsFromOtherMapping(idx_t map_idx, DataChunk &input,
+                                                                     DataChunk &intermediate, DataChunk &out) const {
+	auto &mapping_column = intermediate.data[map_idx];
+
+	SelectionVector sel_vec;
+	sel_vec.Initialize(intermediate.size());
+	// Create a selection vector that maps from output row -> input row
+	UnifiedVectorFormat mapping_vector_data;
+	mapping_column.ToUnifiedFormat(intermediate.size(), mapping_vector_data);
+	auto mapping_data = (sel_t *)mapping_vector_data.data;
+	for (idx_t i = 0; i < intermediate.size(); i++) {
+		// The index in the input column that produced this output tuple
+		idx_t idx = mapping_vector_data.sel->get_index(i);
+		const auto input_row = mapping_data[idx];
+		D_ASSERT(input_row < STANDARD_VECTOR_SIZE);
+		sel_vec.set_index(i, input_row);
+	}
+
+	// Add the projected columns, and apply the selection vector
+	for (idx_t project_idx = 0; project_idx < projected_input.size(); project_idx++) {
+		auto source_idx = projected_input[project_idx];
+		D_ASSERT(source_idx < input.data.size());
+		auto target_idx = map_idx + project_idx;
+
+		auto &target_column = out.data[target_idx];
+		auto &source_column = input.data[source_idx];
+
+		target_column.Slice(source_column, sel_vec, intermediate.size());
+		// Note: we can avoid flattening this
 	}
+}
+
+void PhysicalTableInOutFunction::AddProjectedColumnsFromConstantMapping(idx_t map_idx, DataChunk &input,
+                                                                        DataChunk &intermediate, DataChunk &out) const {
+	auto &mapping_column = intermediate.data[map_idx];
+	D_ASSERT(mapping_column.GetVectorType() == VectorType::CONSTANT_VECTOR);
+
+	UnifiedVectorFormat mapping_vector_data;
+	mapping_column.ToUnifiedFormat(intermediate.size(), mapping_vector_data);
```

Comment:
`ToUnifiedFormat` seems unnecessary here - can we just use `ConstantVector::GetData<uint32_t>()`?